deeppavlov/models/kbqa/rel_ranking_infer.py

```diff
@@ -56,35 +88,139 @@ def load(self) -> None:
-                                                                        List[List[Tuple[str, Any]]]:
-        rels_with_scores_batch = []
-        for question, candidate_rels in zip(question_batch, candidate_rels_batch):
-            rels_with_scores_batch.append(self.rank_rels(question, candidate_rels))
-        return rels_with_scores_batch
+    def __call__(self, questions_list: List[str],
+                 candidate_answers_list: List[List[Tuple[str]]],
+                 entities_list: List[List[str]] = None,
+                 template_answers_list: List[str] = None) -> List[str]:
+        answers = []
+        confidence = 0.0
+        if entities_list is None:
+            entities_list = [[] for _ in questions_list]
+        if template_answers_list is None:
+            template_answers_list = ["" for _ in questions_list]
+        for question, candidate_answers, entities, template_answer in \
+                zip(questions_list, candidate_answers_list, entities_list, template_answers_list):
+            answers_with_scores = []
+            answer = "Not Found"
+            if self.rank:
+                n_batches = len(candidate_answers) // self.batch_size + int(
+                    len(candidate_answers) % self.batch_size > 0)
+                for i in range(n_batches):
+                    questions_batch = []
+                    rels_batch = []
+                    rels_labels_batch = []
+                    answers_batch = []
+                    entities_batch = []
+                    confidences_batch = []
+                    for candidate_ans_and_rels in candidate_answers[i * self.batch_size: (i + 1) * self.batch_size]:
+                        candidate_rels = []
+                        if candidate_ans_and_rels:
+                            candidate_rels = candidate_ans_and_rels["relations"]
+                            candidate_rels = [candidate_rel.split('/')[-1] for candidate_rel in candidate_rels]
+                            candidate_answer = candidate_ans_and_rels["answers"]
+                            candidate_entities = candidate_ans_and_rels["entities"]
+                            candidate_confidence = candidate_ans_and_rels["rel_conf"]
+                            candidate_rels_str = " # ".join([self.rel_q2name[candidate_rel] \
+                                                             for candidate_rel in candidate_rels if
+                                                             candidate_rel in self.rel_q2name])
+                        if candidate_rels_str:
+                            questions_batch.append(question)
+                            rels_batch.append(candidate_rels)
+                            rels_labels_batch.append(candidate_rels_str)
+                            answers_batch.append(candidate_answer)
+                            entities_batch.append(candidate_entities)
+                            confidences_batch.append(candidate_confidence)
```

Comment:
Could candidate_ans_and_rels be False and candidate_rels_str be True? In this case we will reference variables before assignment